core/embed/prodtest/main.c

```diff
@@ -513,9 +603,443 @@ static void test_otp_write_device_variant(const char *args) {
+    return;
+  }
+
+  size_t offset = 0;
+  if (cert[0] == 0xC0) {
+    // TLS identity certificate chain.
+    size_t tls_identity_size = (cert[1] << 8) + cert[2];
+    size_t cert_chain_size = (cert[3] << 16) + (cert[4] << 8) + cert[5];
+    size_t first_cert_size = (cert[6] << 16) + (cert[7] << 8) + cert[8];
+    if (tls_identity_size + 3 > cert_size ||
+        cert_chain_size + 3 > tls_identity_size ||
+        first_cert_size > cert_chain_size) {
+      vcp_println("ERROR invalid TLS identity in 0x%04x.", oid);
+      return;
+    }
+    offset = 9;
+    cert_size = first_cert_size;
+  }
+
+  if (cert_size == 0) {
+    vcp_println("ERROR no certificate in 0x%04x.", oid);
+    return;
+  }
+
+  vcp_print("OK ");
+  vcp_println_hex(cert + offset, cert_size);
+}
+
+static void cert_write(uint16_t oid, char *data) {
+  // Enable writing to the certificate slot.
+  optiga_metadata metadata = {0};
+  metadata.change = OPTIGA_ACCESS_ALWAYS;
+  set_metadata(oid, &metadata);  // Ignore result.
+
+  uint8_t data_bytes[1024];
+
+  int len = get_from_hex(data_bytes, sizeof(data_bytes), data);
+  if (len < 0) {
+    vcp_println("ERROR Hexadecimal decoding error %d.", len);
+    return;
+  }
+
+  optiga_result ret = optiga_set_data_object(oid, false, data_bytes, len);
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_set_data error %d for 0x%04x.", ret, oid);
+    return;
+  }
+
+  vcp_println("OK");
+}
+
+static void pubkey_read(uint16_t oid) {
+  // Enable key agreement usage.
+
+  optiga_metadata metadata = {0};
+  uint8_t key_usage = OPTIGA_KEY_USAGE_KEYAGREE;
+  metadata.key_usage.ptr = &key_usage;
+  metadata.key_usage.len = 1;
+  metadata.execute = OPTIGA_ACCESS_ALWAYS;
+
+  if (!set_metadata(oid, &metadata)) {
+    return;
+  }
+
+  // Execute ECDH with base point to get the x-coordinate of the public key.
+  static const uint8_t BASE_POINT[] = {
+      0x03, 0x42, 0x00, 0x04, 0x6b, 0x17, 0xd1, 0xf2, 0xe1, 0x2c, 0x42, 0x47,
+      0xf8, 0xbc, 0xe6, 0xe5, 0x63, 0xa4, 0x40, 0xf2, 0x77, 0x03, 0x7d, 0x81,
+      0x2d, 0xeb, 0x33, 0xa0, 0xf4, 0xa1, 0x39, 0x45, 0xd8, 0x98, 0xc2, 0x96,
+      0x4f, 0xe3, 0x42, 0xe2, 0xfe, 0x1a, 0x7f, 0x9b, 0x8e, 0xe7, 0xeb, 0x4a,
+      0x7c, 0x0f, 0x9e, 0x16, 0x2b, 0xce, 0x33, 0x57, 0x6b, 0x31, 0x5e, 0xce,
+      0xcb, 0xb6, 0x40, 0x68, 0x37, 0xbf, 0x51, 0xf5};
+  uint8_t public_key[32] = {0};
+  size_t public_key_size = 0;
+  optiga_result ret = optiga_calc_ssec(
+      OPTIGA_CURVE_P256, OID_KEY_DEV, BASE_POINT, sizeof(BASE_POINT),
+      public_key, sizeof(public_key), &public_key_size);
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_calc_ssec error %d.", ret);
+    return;
+  }
+
+  vcp_print("OK ");
+  vcp_println_hex(public_key, public_key_size);
+}
+
+static void keyfido_write(char *data) {
+  static const size_t EPH_PUB_KEY_SIZE = 33;
+  static const size_t PAYLOAD_SIZE = 32;
+  static const size_t CIPHERTEXT_OFFSET = EPH_PUB_KEY_SIZE;
```

Comment:
Is it intentionally defined as static?

Reply:
Makes no difference, does it?

Reply:
There is no difference in terms of the result, but there can be a difference in terms of the resulting binary code. Since I rarely see such a construction, I am curious if there is any motivation behind it.

Reply:
No motivation.